### ISSUE TITLE
[Feature] Outcome aware train-test split

### ISSUE THREAD

**Issue by egillax** [OHDSI/PatientLevelPrediction#396]
According to [this](https://www.sciencedirect.com/science/article/pii/S1386505622000764) paper, over a certain number of outcome events the discriminative model performance stops improving for L1 logistic regression at least. 

This could be taken advantage of when the data size is very big to limit the training set to that number (or slightly above to be safe) and move the rest of the data to the test set.  `splitData` [takes](https://github.com/OHDSI/PatientLevelPrediction/blob/main/R/RunPlp.R#L295) the `population` as an input. So should be relatively easy to adjust the splits based on # of outcome events.
